include/deal.II/base/table.h

```diff
@@ -2367,6 +2624,168 @@ Table<2,T>::n_cols () const
 
 
 
+//---------------------------------------------------------------------------
+namespace TransposeTableIterators
+{
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase()
+    :
+    container(nullptr),
+    linear_index(std::numeric_limits<decltype(linear_index)>::max())
+  {}
+
+
+
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase(const container_pointer_type table)
+    :
+    container(table),
+    linear_index(container->values.size())
+  {}
+
+
+
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase(const AccessorBase<T, false> &a)
+    :
+    container(a.container),
+    linear_index(a.linear_index)
+  {}
+
+
+
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase(const container_pointer_type table,
+                                           const std::ptrdiff_t         index)
+    :
+    container(table),
+    linear_index(index)
+  {}
+
+
+
+  template <typename T, bool Constness>
+  inline
+  const T &
+  AccessorBase<T, Constness>::value() const
+  {
+    Assert(0 <= this->linear_index,
+           ExcMessage("The current iterator points outside of the table."));
+    return this->container->values[this->linear_index];
+  }
+
+
+
+  template <typename T, bool Constness>
+  inline
+  typename AccessorBase<T, Constness>::size_type
+  AccessorBase<T, Constness>::AccessorBase::row() const
+  {
+    Assert(!container->empty(), ExcMessage("An empty table has no rows or columns."));
+    return linear_index % container->n_rows();
```

Comment:
same here and in all other places, obviously -- I mean, you do access `n_rows()`, so couldn't you also make sure that `linear_index` is within range?

Reply:
Good point. I'll add a check.